lib/human_readable_id.rb

```diff
@@ -0,0 +1,26 @@
+class HumanReadableId
+  ID_LENGTH = 8
+  MAX_NUMBER = Base32::Crockford.decode('Z' * ID_LENGTH) # Largest 8 digit number
+
+  def self.update_unique_id(klass, primary_key, column)
+    ActiveRecord::Base.transaction(requires_new: true) do
```

Comment:
No, it doesn't require an outer transaction. It just ensures it is contained within a transaction. I don't think it is necessary to add a comment as it is explained well in the Rails docs.